app/src/main/java/com/example/hw4/adapter/NoteListAdapter.kt

```diff
@@ -0,0 +1,38 @@
+package com.example.hw4.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import com.example.hw4.data.Note
+import com.example.hw4.databinding.ItemBinding
+
+
+class NoteListAdapter(private val itemClick: (Note) -> Unit) : ListAdapter<Note, NoteViewHolder>(
+    DIFF_CALLBACK
+) {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NoteViewHolder {
+        return NoteViewHolder(
+            binding = ItemBinding.inflate(LayoutInflater.from(parent.context), parent, false),
+            itemClick = itemClick
+        )
+    }
+
+    override fun onBindViewHolder(holder: NoteViewHolder, position: Int) {
+        holder.bind(getItem(position))
+    }
+
+
+    companion object {
+        private val DIFF_CALLBACK = object : DiffUtil.ItemCallback<Note>() {
+            override fun areItemsTheSame(oldItem: Note, newItem: Note): Boolean {
+                return oldItem == newItem
+            }
```

Comment:
это верхнеуровневая проверка элементов по первичному признаку, здесь должно быть что-то вроде
```suggestion
            override fun areItemsTheSame(oldItem: Note, newItem: Note): Boolean {
                return oldItem.id == newItem.id
            }
```